#159361816 sort search results

ozimos opened the pull request:
#### What does this PR do?
Adds the ability to sort the search results for product

#### Description of Task to be completed?
Allow the user to sort the results fo their product search by the following criteria
- Price
- Vendor Name
- Date Created

Add searchbox to the navbar
#### How should this be manually tested?
View the review app

#### Any background context you want to provide?
NA
#### What are the relevant pivotal tracker stories?
#159361816

#### Screenshots or gifs (if appropriate):
![sort2](https://user-images.githubusercontent.com/27624208/44043072-d5f8811c-9f19-11e8-909d-3f03f508982c.gif)

#### Questions:

otseobande reviewed (approved):
Nice implementation of the `sort search results` feature.

nouwatinjacob reviewed (approved):
Good job. Keep it up

ozimos committed:
feat(sort) add sort to search workflow
- allows users to sort their product search results by
- vendor
- price
- date created
[Finishes #159361816]
